codegen/cx1/test3d.cpp

```diff
@@ -0,0 +1,933 @@
+/*  Copyright (C) 2015 Imperial College London and others.
+ *
+ *  Please see the AUTHORS file in the main source directory for a
+ *  full list of copyright holders.
+ *
+ *  Gerard Gorman
+ *  Department of Earth Science and Engineering
+ *  Imperial College London
+ *
+ *  g.gorman@imperial.ac.uk
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *  notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above
+ *  copyright notice, this list of conditions and the following
+ *  disclaimer in the documentation and/or other materials provided
+ *  with the distribution.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ *  CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ *  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS
+ *  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ *  TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ *  TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ *  THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ *  SUCH DAMAGE.
+ */
+
+/* Test case implements:
+
+   Graves, Robert W. "Simulating seismic wave propagation in 3D
+   elastic media using staggered-grid finite differences." Bulletin of
+   the Seismological Society of America 86.4 (1996): 1091-1106.
+
+*/
+
+
+// #include "opesciIO.h"
```

Comment:
regenerated the file
also added //#define M_PI 3.14159265358979323846 because intel compiler on windows doesn't have this macro in cmath, and I was testing on windows at home